book/src/asm/functions.md

```diff
@@ -0,0 +1,74 @@
+# Functions
+
+Machine functions are the entry points to this machine. They can be called from another machine or from the outside.
+
+For static machines, functions simply indicate which columns should be exposed as inputs and outputs. They do not contain statements, since the business logic of all functions in the static machine is defined directly in constraints. We refer to the previous example [here](./machines.md#static-machines).
+
+From now on, we describe functions in dynamic machines with the example of this simple machine:
+
+```
+{{#include ../../../test_data/asm/book/function.asm:all}}
+```
+
+## Function inputs and outputs
+
+> For dynamic machines, function inputs and outputs are not supported yet
+
+## Expressions
+
+### Field element literals
+
+Field element literals are signed elements of the prime field.
+
+```
+{{#include ../../../test_data/asm/book/function.asm:literals}}
+```
+
+### Registers
+
+Registers can be used as expressions, with the exception of assignment registers.
```

Comment:
Hmm is that really true?

Reply:
What I mean is: can assignment registers really not be used as expressions? Like

```
A <== X
```
where X is assgn